pallets/pips/src/lib.rs

```diff
@@ -1612,3 +1569,20 @@ impl<T: Trait> Module<T> {
         proposal_data
     }
 }
+
+/// Move `slice[old]` to `slice[new]`, shifting other elements as necessary.
+///
+/// For example, given `let arr = [1, 2, 3, 4, 5];`,
+/// then `move_element(&mut arr, 1, 3);` would result in `[1, 3, 4, 2, 5]`
+/// whereas `move_element(&mut arr, 3, 1)` would yield `[1, 4, 2, 3, 5]`.
+fn move_element<T: Copy>(slice: &mut [T], old: usize, new: usize) {
```

Comment:
@Centril Do you think we could refactor that to use `Vec::swap`? I would like to avoid copies.

Reply:
Using `Vec::swap`s in a loop doesn't actually entail no copies. It will still perform copies, but with bounds checks in every loop iteration (tested it). When I compared the x86 ASM/LLVM bitcode, the current solution was the safe one that came up on top in terms of codegen.